src/__tests__/WeekView.test.js

```diff
@@ -1,10 +1,454 @@
+      color: colors[index],
+    }));
+    const { queryAllByHintText } = render(
+      /* eslint-disable-next-line react/jsx-props-no-spreading */
+      <WeekView events={events} {...props} />,
+    );
+    queryAllByHintText(/Show event \d+/).forEach((renderedEvent) => {
+      const eventId = Number(
+        extractEventIdFromTestID(renderedEvent.props.testID),
+      );
+      const expectedColor = events.find((evt) => evt.id === eventId).color;
+      expect(renderedEvent).toHaveStyle({ backgroundColor: expectedColor });
+    });
+  };
+
+  describe('with non-overlapping events', () => {
+    const baseEventsNoOverlap = [
+      {
+        id: 1,
+        startDate: new Date(2021, 4, 2, 12, 0),
+        endDate: new Date(2021, 4, 2, 13, 0),
+      },
+      {
+        id: 2,
+        startDate: new Date(2021, 4, 2, 13, 0),
+        endDate: new Date(2021, 4, 2, 13, 45),
+      },
+      {
+        id: 7,
+        startDate: new Date(2021, 4, 3, 11, 14, 56),
+        endDate: new Date(2021, 4, 3, 11, 30, 21),
+      },
+      {
+        id: 19,
+        startDate: new Date(2021, 4, 4, 13, 32),
+        endDate: new Date(2021, 4, 4, 14, 35),
+      },
+      {
+        id: 52,
+        startDate: new Date(2021, 4, 5, 18, 23),
+        endDate: new Date(2021, 4, 5, 19, 57),
+      },
+    ].map((evt) => ({ ...evt, description: `event ${evt.id}` }));
+
+    const baseProps = {
+      numberOfDays: 5,
+      selectedDate: new Date(2021, 4, 1),
+      hoursInDisplay: 10,
+      startHour: 11,
+    };
+
+    it('shows all events', () =>
+      runTestShowsAll(baseEventsNoOverlap, baseProps));
+
+    it('shows each event with its color', () => {
+      const colors = ['blue', 'lightblue', 'purple', 'yellow', 'pink'];
+      runTestShowColor(baseEventsNoOverlap, baseProps, colors);
+    });
+  });
+
+  describe('with overlapped events', () => {
+    const baseEventsOverlap = [
+      {
+        id: 9,
+        startDate: new Date(2020, 7, 24, 17),
+        endDate: new Date(2020, 7, 24, 18),
+      },
+      {
+        id: 43,
+        startDate: new Date(2020, 7, 24, 17),
+        endDate: new Date(2020, 7, 24, 18),
+      },
+      {
+        id: 11,
+        startDate: new Date(2020, 7, 25, 12, 15),
+        endDate: new Date(2020, 7, 25, 14, 30),
+      },
+      {
+        id: 2,
+        startDate: new Date(2020, 7, 25, 13, 0),
+        endDate: new Date(2020, 7, 25, 13, 45),
+      },
+      {
+        id: 3,
+        startDate: new Date(2020, 7, 24, 17),
+        endDate: new Date(2020, 7, 24, 18),
+      },
+      {
+        id: 1,
+        startDate: new Date(2020, 7, 26, 13),
+        endDate: new Date(2020, 7, 26, 20),
+      },
+    ].map((evt) => ({ ...evt, description: `event ${evt.id}` }));
+
+    const baseProps = {
+      numberOfDays: 3,
+      selectedDate: new Date(2020, 7, 24),
+      hoursInDisplay: 12,
+      startHour: 12,
+    };
+
+    it('shows all events', () => runTestShowsAll(baseEventsOverlap, baseProps));
```

Comment:
I'm thinking if we can use https://jestjs.io/docs/api#testeachtablename-fn-timeout.

Reply:
Great idea! it looks much better now